pkg/operator/status_test.go

```diff
@@ -176,3 +178,64 @@ func TestIsControllerConfigCompleted(t *testing.T) {
 		})
 	}
 }
+
+type mockMCPLister struct{}
+
+func (mcpl *mockMCPLister) List(selector labels.Selector) (ret []*mcfgv1.MachineConfigPool, err error) {
+	return nil, nil
+}
+func (mcpl *mockMCPLister) Get(name string) (ret *mcfgv1.MachineConfigPool, err error) {
+	return nil, nil
+}
+
+type mockClusterOperatorsClient struct{
+	co *configv1.ClusterOperator
+}
+
+func (coc *mockClusterOperatorsClient) Create(co *configv1.ClusterOperator) (*configv1.ClusterOperator, error) {
+	return coc.co, nil
+}
+func (coc *mockClusterOperatorsClient) UpdateStatus(co *configv1.ClusterOperator) (*configv1.ClusterOperator, error) {
+	return coc.co, nil
+}
+func (coc *mockClusterOperatorsClient) Get(name string, options metav1.GetOptions) (*configv1.ClusterOperator, error) {
+	return coc.co, nil
+}
+
+func TestOperatorFailingStatusClearsOut(t *testing.T) {
```

Comment:
this test fail correctly w/o the patch to fix the bug

Reply:
I'll also add the other tests for the other conditions to make sure we never regress anymore on this...